pkg/validation_fuzz_test.go

```diff
@@ -0,0 +1,174 @@
+	f.Add("org.springframework.boot", "spring-boot-starter", "2.7.0", "import", "pom")
+	f.Add("", "", "", "", "")
+	f.Add("com.example;rm -rf /", "artifact", "1.0", "test", "jar")
+	f.Add("a", "b", "c", "d", "e")
+	f.Add("com.example", "artifact", "1.0.0<script>alert('xss')</script>", "compile", "jar")
+	
+	f.Fuzz(func(t *testing.T, groupID, artifactID, version, scope, patchType string) {
+		patch := Patch{
+			GroupID:    groupID,
+			ArtifactID: artifactID,
+			Version:    version,
+			Scope:      scope,
+			Type:       patchType,
+		}
+		
+		// The function should not panic on any input
+		err := ValidatePatch(patch)
+		
+		// If no error, verify the patch meets basic requirements
+		if err == nil {
+			// These should never be empty if validation passed
+			if groupID == "" || artifactID == "" || version == "" {
+				t.Errorf("ValidatePatch allowed empty required field: groupID=%q, artifactID=%q, version=%q", groupID, artifactID, version)
+			}
+			
+			// Check scope is valid if present
+			if scope != "" {
+				validScopes := map[string]bool{
+					"compile": true, "provided": true, "runtime": true,
+					"test": true, "system": true, "import": true,
+				}
+				if !validScopes[scope] {
+					t.Errorf("ValidatePatch allowed invalid scope: %q", scope)
+				}
+			}
+			
+			// Check length limits
+			if len(groupID) > 256 || len(artifactID) > 256 || len(version) > 128 {
+				t.Error("ValidatePatch allowed values exceeding length limits")
+			}
+			
+			// Type length limit
+			if patchType != "" && len(patchType) > 64 {
+				t.Errorf("ValidatePatch allowed type exceeding length limit: %d chars", len(patchType))
+			}
+		}
+	})
+}
+
+// FuzzValidatePropertyPatch tests ValidatePropertyPatch with random inputs
+func FuzzValidatePropertyPatch(f *testing.F) {
+	// Add seed corpus
+	f.Add("spring.version", "5.3.23")
+	f.Add("", "")
+	f.Add("prop", "value")
+	f.Add("my.property", "my value with spaces")
+	f.Add("prop", "<script>alert('xss')</script>")
+	f.Add("prop", "value\n<malicious>tag</malicious>")
+	f.Add("very.long.property.name", "very long value")
+	
+	f.Fuzz(func(t *testing.T, property, value string) {
+		// The function should not panic on any input
+		err := ValidatePropertyPatch(property, value)
+		
+		// If no error, verify the values meet requirements
+		if err == nil {
+			if property == "" || value == "" {
+				t.Errorf("ValidatePropertyPatch allowed empty values: property=%q, value=%q", property, value)
+			}
+			
+			if len(property) > 256 || len(value) > 1024 {
+				t.Error("ValidatePropertyPatch allowed values exceeding length limits")
+			}
+			
+			// Check for XML injection patterns that should be rejected
+			if containsXMLInjection(value) {
+				t.Errorf("ValidatePropertyPatch allowed value with XML injection: %q", value)
+			}
+		}
+	})
+}
+
+// FuzzValidateFilePath tests ValidateFilePath with random inputs
+func FuzzValidateFilePath(f *testing.F) {
+	// Add seed corpus
+	f.Add("test.xml")
+	f.Add("/absolute/path/file.xml")
```

Comment:
I'm curious if we have seen any issues with this. I don't recall having problems with the paths as those are already known.